src/github.ts

```diff
@@ -546,16 +546,42 @@ export class GitHub {
   /**
    * Iterate through merged pull requests with a max number of results scanned.
    *
-   * @param {number} maxResults maxResults - Limit the number of results searched.
-   *   Defaults to unlimited.
+   * @param {string} targetBranch The base branch of the pull request
+   * @param {string} status The status of the pull request
+   * @param {number} maxResults Limit the number of results searched. Defaults to
+   *   unlimited.
+   * @param {boolean} includeFiles Whether to fetch the list of files included in
+   *   the pull request. Defaults to `true`.
    * @yields {PullRequest}
    * @throws {GitHubAPIError} on an API error
    */
   async *pullRequestIterator(
+    targetBranch: string,
+    status: 'OPEN' | 'CLOSED' | 'MERGED' = 'MERGED',
+    maxResults: number = Number.MAX_SAFE_INTEGER,
+    includeFiles = true
```

Comment:
nit: I would be tempted to pull this into an interface, so that it's more clear what each field means when invoking:

```
pullRequestIterator(opts: PullRequestOpts = {targetBranch: 'main', status: 'MERGED', maxResults: Number.MAX_SAFE_INTEGER, includeFiles: true});
```